### DISCUSSION TITLE
Add slow parameter to the cover plaform

### DISCUSSION THREAD

**Discussion by tetienne** [home-assistant/architecture#602] [Entity Models]
## Context
Some cover like the io:RollerShutterWithLowSpeedManagementIOComponent from Somfy support two modes when they move: the normal one and the slow one (silent).

## Proposal

* Add to `set_cover_position` and `async_set_cover_position` an optional boolean _slow_ parameter. 
* Create a flag SUPPORT_SLOW_SPEED

## Consequences

It won't change anything for the existing cover platforms. It will avoid to create a dedicated service for this feature. From an end user perspective, it will be possible from the UI to choose how to move the covers (from a simple or double chevron icon for instance). It will also allow to create automation to move quietly the covers during the night or the morning for instance.

**Comment by tetienne** [comment]
As a workaround currently, we have to use an entity switch to enable or disable the slow/silent mode.

**Comment by tetienne** [comment]
Here an example of such option switch: https://github.com/iMicknl/ha-tahoma/pull/497

**Comment by Jc2k** [comment]
Hi. Not really my area, but I think the thing that reviewers normally ask for is to review other cover implementations and find out:

* Is this a common feature or specific to RollerShutterWithLowSpeedManagementIOComponent.
* If it's common what range of values do they take? In particular we probably don't want to repeat the pain we had with fans where we almost immediately didn't have enough "blessed" speeds.
* Do any of the "bridges" (that let Google Asssistant, Alexa, or Siri/HomeKit) implement this? Is your API compatible with them. If they don't implement this, will we just not send a speed and let the integration decide what to do?

If you have answers for these it might help anyone reviewing the suggestion.

**Comment by bdraco** [comment]
This looks like a subset of what is being discussed in https://github.com/home-assistant/architecture/discussions/789
